### ISSUE TITLE
Annotation for automatic optional initializers

### ISSUE THREAD

**Issue by R4PaSs** [nitlang/nit#1843]
Hi,

Recently, @BlackMinou and I had a discussion about how to have an attribute within a class with a default value whilst still having the possibility of assigning a specific value at construction-time.

Basically, what I have advised is something along the lines of this

~~~nit
class A
    var x: Int is noautoinit
    protected fun set_x(x: nullable Int) is autoinit do self.x = x or else -1
end

var a = new A # Creates a new instance of A with A::x == -1
var b = new A(4) # Creates a new instance of A with A::x == 4
~~~

Which, and @BlackMinou will agree, is quite verbose.

After discussing with @privat, since this is quite a special, and in my opinion common, case, a new annotation could do the job for us.

What we have come up with is this

~~~nit
class A
    var x: Int = -1 is optional
end

var a = new A # Creates a new instance of A with A::x == -1
var b = new A(4) # Creates a new instance of A with A::x == 4
~~~

I think this is quite elegant, and @privat seems to think the implementation will not be hard.

Is everyone on board for this or are there some drawbacks we failed to see ?

**Comment by privat** [comment]
I do not mind adding new annotations to the zoo if people plan to pet them

**Comment by privat** [comment]
closed by #1965
